iroh-net/src/magicsock.rs

```diff
@@ -1176,6 +1176,27 @@ impl Actor {
         out
     }
 
+    /// Split a number of transmits into individual packets.
+    ///
+    /// For each transmit, if it has a segment size, it will be split into
+    /// multiple packets according to that segment size. If it does not have a
+    /// segment size, the contents will be sent as a single packet.
+    fn split_packets(transmits: Vec<quinn_udp::Transmit>) -> Vec<Bytes> {
+        let mut res = Vec::with_capacity(transmits.len());
+        for transmit in transmits {
+            if let Some(segment_size) = transmit.segment_size {
+                let len = transmit.contents.len();
+                for min in (0..len).step_by(segment_size) {
+                    let max = (min + segment_size).min(len);
+                    res.push(transmit.contents.slice(min..max));
```

Comment:
isn't this a lot easier using `transmit.contents.chunks(segment_size)`?

Reply:
It might be that I just reimplemented that one...

Reply:
it's not quite as straightforward if you want to use the Bytes in a 0 copy way, but I pushed an update.